services/sts/src/main/java/software/amazon/awssdk/services/sts/auth/StsCredentialsProvider.java

```diff
@@ -83,6 +96,14 @@ public void close() {
         sessionCache.close();
     }
 
+    public Duration staleTime() {
```

Comment:
Can we add javadoc: The amount of time, relative to STS token expiration, that the cached credentials are considered stale and should no longer be used. All threads will block until the value is updated.

Reply:
Done